exit with error even errors are below severity level

i execute pint on a specific folder and with `--min-severity=fatal`

```
pint lint ./apps/alertmanager-config/charts/rules --min-severity=fatal

level=INFO msg="Finding all rules to check" paths=["./apps/alertmanager-config/charts/rules"]
level=INFO msg="Checking Prometheus rules" entries=789 workers=10 online=true
level=INFO msg="Problems found" Bug=33 Warning=46 Information=6
level=INFO msg="85 problem(s) not visible because of --min-severity=fatal flag"
level=ERROR msg="Execution completed with error(s)" err="found 33 problem(s) with severity Bug or higher"
Error: Process completed with exit code 1.
```

i expect no error here as there are no fatal errors

--- a/cmd/pint/lint.go
+++ b/cmd/pint/lint.go
@@ -119,6 +119,18 @@ func actionLint(ctx context.Context, c *cli.Command) error {
 	if err != nil {
 		return fmt.Errorf("invalid --%s value: %w", failOnFlag, err)
 	}
+	if minSeverity > failOn {
+		slog.LogAttrs(
+			ctx,
+			slog.LevelWarn,
+			fmt.Sprintf(
+				"You have --%s set to a higher severity value than --%s, pint might exit with a non-zero code but you won't see the problem that caused it",
+				minSeverityFlag, failOnFlag,
+			),
+			slog.String(minSeverityFlag, minSeverity.String()),
+			slog.String(failOnFlag, failOn.String()),
+		)
+	}
 
 	reps := []reporter.Reporter{}
 	if c.Bool(teamCityFlag) {

--- a/cmd/pint/tests/0128_lint_fail_on_warning_only.txt
+++ b/cmd/pint/tests/0128_lint_fail_on_warning_only.txt
@@ -12,6 +12,7 @@ groups:
 -- stderr.txt --
 level=INFO msg="Finding all rules to check" paths=["rules"]
 level=INFO msg="Checking Prometheus rules" entries=1 workers=10 online=true
+level=WARN msg="You have --min-severity set to a higher severity value than --fail-on, pint might exit with a non-zero code but you won't see the problem that caused it" min-severity=Bug fail-on=Warning
 level=INFO msg="Problems found" Warning=1
 level=INFO msg="1 problem(s) not visible because of --min-severity=bug flag"
 level=ERROR msg="Execution completed with error(s)" err="found 1 problem(s) with severity Warning or higher"

--- a/cmd/pint/tests/0220_lint_min_severity_fatal_ok.txt
+++ b/cmd/pint/tests/0220_lint_min_severity_fatal_ok.txt
@@ -1,0 +1,38 @@
+! exec pint --no-color lint --min-severity=fatal rules
+! stdout .
+cmp stderr stderr.txt
+
+-- stderr.txt --
+level=INFO msg="Loading configuration file" path=.pint.hcl
+level=INFO msg="Finding all rules to check" paths=["rules"]
+level=INFO msg="Checking Prometheus rules" entries=3 workers=10 online=true
+level=WARN msg="You have --min-severity set to a higher severity value than --fail-on, pint might exit with a non-zero code but you won't see the problem that caused it" min-severity=Fatal fail-on=Bug
+level=INFO msg="Problems found" Bug=1 Warning=1 Information=1
+level=INFO msg="3 problem(s) not visible because of --min-severity=fatal flag"
+level=ERROR msg="Execution completed with error(s)" err="found 1 problem(s) with severity Bug or higher"
+-- rules/0001.yml --
+groups:
+- name: foo
+  rules:
+  - alert: info
+    expr: info == 0
+  - alert: warning
+    expr: warning == 0
+  - alert: bug
+    expr: bug == 0
+
+-- .pint.hcl --
+rule {
+  selector "info" {
+    requiredLabels = ["xxx"]
+    severity = "info"
+  }
+  selector "warning" {
+    requiredLabels = ["xxx"]
+    severity = "warning"
+  }
+  selector "bug" {
+    requiredLabels = ["xxx"]
+    severity = "bug"
+  }
+}

--- a/cmd/pint/tests/0221_lint_min_severity_fatal_err.txt
+++ b/cmd/pint/tests/0221_lint_min_severity_fatal_err.txt
@@ -1,0 +1,49 @@
+! exec pint --no-color lint --min-severity=fatal rules
+! stdout .
+cmp stderr stderr.txt
+
+-- stderr.txt --
+level=INFO msg="Loading configuration file" path=.pint.hcl
+level=INFO msg="Finding all rules to check" paths=["rules"]
+level=INFO msg="Checking Prometheus rules" entries=4 workers=10 online=true
+level=WARN msg="You have --min-severity set to a higher severity value than --fail-on, pint might exit with a non-zero code but you won't see the problem that caused it" min-severity=Fatal fail-on=Bug
+Fatal: required matcher missing (promql/selector)
+  ---> rules/0001.yml:11 -> `fatal`
+11 |     expr: fatal == 0
+               ^^^^^ This vector selector must specify `xxx` label. Please add a `{xxx="..."} matcher.
+
+level=INFO msg="Problems found" Fatal=1 Bug=1 Warning=1 Information=1
+level=INFO msg="3 problem(s) not visible because of --min-severity=fatal flag"
+level=ERROR msg="Execution completed with error(s)" err="found 2 problem(s) with severity Bug or higher"
+-- rules/0001.yml --
+groups:
+- name: foo
+  rules:
+  - alert: info
+    expr: info == 0
+  - alert: warning
+    expr: warning == 0
+  - alert: bug
+    expr: bug == 0
+  - alert: fatal
+    expr: fatal == 0
+
+-- .pint.hcl --
+rule {
+  selector "info" {
+    requiredLabels = ["xxx"]
+    severity = "info"
+  }
+  selector "warning" {
+    requiredLabels = ["xxx"]
+    severity = "warning"
+  }
+  selector "bug" {
+    requiredLabels = ["xxx"]
+    severity = "bug"
+  }
+  selector "fatal" {
+    requiredLabels = ["xxx"]
+    severity = "fatal"
+  }
+}

